lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -1416,26 +1416,44 @@ class LineChartPainter extends AxisChartPainter<LineChartData> {
   }
 
   /// find the nearest spot base on the touched offset
-  LineBarSpot? _getNearestTouchedSpot(Size viewSize, Offset touchedPoint, LineChartBarData barData,
-      int barDataPosition, PaintHolder<LineChartData> holder) {
+  LineBarSpot? _getNearestTouchedSpot(
+      Size viewSize,
+      Offset touchedPoint,
+      LineChartBarData barData,
+      int barDataPosition,
+      PaintHolder<LineChartData> holder) {
     final data = holder.data;
     if (!barData.show) {
       return null;
     }
 
     final chartViewSize = getChartUsableDrawSize(viewSize, holder);
 
-    /// Find the nearest spot (on X axis)
-    for (var i = 0; i < barData.spots.length; i++) {
-      final spot = barData.spots[i];
-      if (spot.isNotNull()) {
-        if ((touchedPoint.dx - getPixelX(spot.x, chartViewSize, holder)).abs() <=
-            data.lineTouchData.touchSpotThreshold) {
-          return LineBarSpot(barData, barDataPosition, spot);
+    final sortedSpots = <FlSpot>[];
+    double? smallestDistance;
+
+    for (var e in barData.spots) {
```

Comment:
Everything is well implemented and working great.
I have only one minor comment. `e` is not descriptive enough. please rename it to something like: `spot`.

Thanks!